lib/Sema/CSSimplify.cpp

```diff
@@ -5674,6 +5674,19 @@ static bool isForKeyPathSubscript(ConstraintSystem &cs,
   return false;
 }
 
+static bool isForKeyPathSubscriptWithoutLabel(ConstraintSystem &cs,
+                                              ConstraintLocator *locator) {
+  if (!locator || !locator->getAnchor())
+    return false;
+
+  if (auto *SE = getAsExpr<SubscriptExpr>(locator->getAnchor())) {
+    auto *indexExpr = dyn_cast<ParenExpr>(SE->getIndex());
```

Comment:
Nit: We don't need `dyn_cast` here, we can `getSemanticsProvidingExpr` directly on `SE->getIndex()` :)